src/core/control/Control.cpp

```diff
@@ -2403,6 +2406,58 @@ auto Control::openFile(fs::path filepath, int scrollToPage, bool forceOpen) -> b
     return true;
 }
 
+auto Control::changePdfBackground() -> bool {
+
+    LoadHandler loadHandler;
+
+    const fs::path current_filepath = ((*this).doc)->getFilepath();
+    const fs::path current_PDFfilepath = ((*this).doc)->getPdfFilepath();
+
+    auto pageNr = getCurrentPageNo();
+    PageRef page = this->doc->getPage(pageNr);
+    Layer* layer = page->getSelectedLayer();
+
+    if (current_PDFfilepath.empty()) {
+        string msg = FS(_F("Error in changing the background: please open a pdf first.")) + loadHandler.getLastError();
+        XojMsgBox::showErrorToUser(getGtkWindow(), msg);
+        return true;
+    }
+
+    if (this->undoRedo->isChanged()) {
+        string msg = FS(_F("Error in changing the background: please save the recent changes first.")) +
+                     loadHandler.getLastError();
+        XojMsgBox::showErrorToUser(getGtkWindow(), msg);
+        return true;
+    }
+
+    if (current_filepath.empty()) {
+        string msg = FS(_F("Error in changing the background: please open and save some file first.")) +
```

Comment:
That formulation can be improved. What about this?
```suggestion
        string msg = FS(_F("Error in changing the background: please save the document first.")) +
```

Reply:
I wrote it like this because the user may try to change the background before having saved. Maybe like this?
`"Error in changing the background: please open (and save) a document first."`
what do you think?

Reply:
I don't agree. It's about saving the document (.xopp-file). You can open a PDF (via `File > Open`) and try to change the PDF background. Then it doesn't make sense if you see the error message "Please open (and save) a document first".

Reply:
You're right, I've changed it to the message you suggested https://github.com/xournalpp/xournalpp/pull/5037#discussion_r1292102884